pkg/verifiedcar/verifiedcar.go

```diff
@@ -33,9 +35,12 @@ var (
 var protoChooser = dagpb.AddSupportToChooser(basicnode.Chooser)
 
 type Config struct {
-	Root       cid.Cid           // The single root we expect to appear in the CAR and that we use to run our traversal against
-	AllowCARv2 bool              // If true, allow CARv2 files to be received, otherwise strictly only allow CARv1
-	Selector   selector.Selector // The selector to execute, starting at the provided Root, to verify the contents of the CAR
+	Root               cid.Cid        // The single root we expect to appear in the CAR and that we use to run our traversal against
+	AllowCARv2         bool           // If true, allow CARv2 files to be received, otherwise strictly only allow CARv1
+	Selector           datamodel.Node // The selector to execute, starting at the provided Root, to verify the contents of the CAR
+	AllowDuplicatesIn  bool           // Handles whether the incoming stream has duplicates
```

Comment:
```suggestion
	ExpectDuplicatesIn  bool           // Handles whether the incoming stream has duplicates
```

I don't think this is optional - if it doesn't have duplicates then I think the traversal code below will bork, so it's now an allowance, it's a strict expectation